generator/src/googleapis/codegen/languages/java/2.0.0/templates/___package___/___api_className___.java.tmpl

```diff
@@ -28,9 +28,13 @@ public class {{ api.className }} extends com.google.api.client.googleapis.servic
   // Note: Leave this static initializer at the top of the file.
   static {
     com.google.api.client.util.Preconditions.checkState(
+        (com.google.api.client.googleapis.GoogleUtils.MAJOR_VERSION == 1 &&
+        (com.google.api.client.googleapis.GoogleUtils.MINOR_VERSION >= 32 ||
+        (com.google.api.client.googleapis.GoogleUtils.MINOR_VERSION == 31 &&
+        com.google.api.client.googleapis.GoogleUtils.BUGFIX_VERSION >= 1))) ||
         com.google.api.client.googleapis.GoogleUtils.MAJOR_VERSION == 2,
         "You are currently running with version %s of google-api-client. " +
-        "You need at least version 2.0.0 of google-api-client to run version " +
+        "You need at least versions 1.31.1 or 2.0.0 of google-api-client to run version " +
```

Comment:
"at least versions 1.31.1" includes 2.0.0.

Reply:
It should accept major version 3 and higher as well. (The original logic assumed the major version was always 1)

Reply:
@suztomo Yep, valid point. I added the `versions 1.31.1 or` in case clients might be hesitant to do a major version upgrade for google-api-client. Probably would make sense just to have v1.31.1 cover both, so i'll remove it.